python/main.py

```diff
@@ -19,56 +23,78 @@
     allow_headers=["*"],
 )
 
+class DBConnection:
+    def __init__(self, dbpath="../db/mercari.sqlite3"):
+        try:
+            self.connection = sqlite3.connect(dbpath)
+            self.cursor = self.connection.cursor()
+            with open('../db/items.db') as f:
+                self.cursor.executescript(f.read())
+            self.connection.commit()
+        except Exception as e:
+            sys.exit(e)
+
+    def execute_insert(self, sql, parameter=[]):
+        try:
+            self.cursor.execute(sql, parameter)
+            self.connection.commit()
+        except Exception as e:
+            sys.exit(e)
+
+    def execute_select(self, sql, parameter=[]):
+        try:
+            sql_result = self.cursor.execute(sql, parameter).fetchall()
+            self.connection.commit()
+        except Exception as e:
+            sys.exit(e)
+
+        return sql_result
+
+    def __del__(self):
+        try:
+            self.connection.close()
+        except Exception as e:
+            sys.exit(e)
+
 @app.get("/")
 def root():
     return {"message": "Hello, world!"}
 
 @app.get("/items")
 def get_item():
-    with sqlite3.connect("../db/mercari.sqlite3") as conn:
-        #カーソル
-        cur = conn.cursor()
-        #データの取得
-        cur.execute("SELECT * FROM items")
-        item_obj = cur.fetchall()
-        #変更の反映
-        conn.commit()
-    
+    db = DBConnection()
+    sql = "SELECT * FROM items"
+    item_obj = db.execute_select(sql)
     return item_obj
 
 @app.post("/items")
-def add_item(name: str = Form(...), category: str = Form(...)):
-    with sqlite3.connect("../db/mercari.sqlite3") as conn:
-        #カーソル
-        cur = conn.cursor()
-        #テーブルがなければ作成
-        cur.execute("CREATE TABLE IF NOT EXISTS items (\
-            id INTEGER AUTO_INCREMENT, \
-            name TEXT NOT NULL,\
-            category TEXT NOT NULL\
-            )")
-
-        #受け取ったデータを追加
-        #(?, ?)はプレースホルダ
-        cur.execute("INSERT INTO items (name, category) VALUES (?, ?)", (name, category))
-        #変更の反映
-        conn.commit()
+def add_item(name: str = Form(...), category_id: str = Form(...), image: UploadFile = File(...)):
+    image_title = pathlib.Path(image.filename).stem
+    image_suffix = pathlib.Path(image.filename).suffix
+    image_filename = hashlib.sha256(image_title.encode()).hexdigest() + image_suffix
+    
+    #画像を保存
+    filepath = f"../images/{image_filename}"
```

Comment:
リクエスト
```
curl -X POST \
--url 'http://localhost:9000/items' \
  -F 'name=jacket' \
  -F 'category_id=fashion' \
  -F 'image=@./images/default.jpg'
```
エラーが出ます。
```
  File "./main.py", line 78, in add_item
    with open(filepath, "w+b") as f:
FileNotFoundError: [Errno 2] No such file or directory: '../images/cd6357efdd966de8c0cb2f876cc89ec74ce35f0968e11743987084bd42fb8944.jpg'
```

```suggestion
    filepath = f"./images/{image_filename}"
```
もしくは
```suggestion
    filepath = f"images/{image_filename}"
```
 かなと思います！

Reply:
修正しました。

Reply:
3e612c845a842946ae0baa155ce498563c93aa3bで修正が確認できました！